### ISSUE TITLE
Cannot change node labels

### ISSUE THREAD

**Issue by flavio** [kubernetes-client/python#240]
I'm trying to change the node labels using the [patch_node](https://github.com/kubernetes-incubator/client-python/blob/master/kubernetes/docs/CoreV1Api.md#patch_node) function but I'm failing a similar way to issue https://github.com/kubernetes-incubator/client-python/issues/187.

This is the code I'm using to add a new label to the node named `minikube`:

```python
from pprint import pprint
from kubernetes import client, config

config.load_kube_config()
client.configuration.debug = True

api_instance = client.CoreV1Api()

body = {
    "op": "add",
    "path": "/metadata/labels",
    "value": ["foo", "bar"]}

api_response = api_instance.patch_node("minikube", body)

pprint(api_response)
```
The API server returns with a 200 code but the new label is not added. The python code is issuing a `PATCH` request using the `application/strategic-merge-patch+json` operation.

On the other hand the request being done by `kubectl label nodes minikube foo=bar` uses the `application/merge-patch+json` operation.

The patch operation is changed to `application/strategic-merge-patch+json` when `body` is not a `list`. This was introduced by [this](https://github.com/kubernetes-incubator/client-python/pull/211/files) automated cherry-pick.
I tried to use a `list` instead of a `dict` for body, but the API server then failed to process the request (it got a JSON unmarshal error).

Any hint on how to handle that?

**Comment by codevulture** [comment]
@flavio : 
I am not sure but this might be helpful to debug, According to flow:
1. 
>https://raw.githubusercontent.com/kubernetes-incubator/client-python/master/kubernetes/client/apis/core_v1_api.py
           `def patch_node_with_http_info`
           `header_params['Content-Type'] = self.api_client.\
            select_header_content_type(['application/json-patch+json', 'application/merge-patch+json',   'application/strategic-merge-patch+json'])`

2)
It selects first one:
> https://github.com/kubernetes-client/python-base/blob/7c8e59b58d9a769f1ca3f818aa91c8a6538b51b1/api_client.py#L489

3)
>https://github.com/kubernetes-client/python-base/blob/7c8e59b58d9a769f1ca3f818aa91c8a6538b51b1/api_client.py#L98 -->  https://github.com/kubernetes-client/python-base/blob/master/rest.py#L156

What if we specify **Header** in request, that might make code to go through this
https://github.com/kubernetes-client/python-base/blob/master/rest.py#L167
branch making request possible.
@mbohlool : Any thoughts?

**Comment by flavio** [comment]
After playing a bit with the source code of this project I realized that using `application/strategic-merge-patch+json` didn't make my initial code work. So I fired up `tcpdump` and I took a closer look at what `kubectl` does.

It turns out it's already possible to manage the labels of a node without doing any changes to this project.
It's only a matter of the `body` of the request.

The following code works like a charm:

```python
from pprint import pprint
from kubernetes import client, config

config.load_kube_config()
client.configuration.debug = True

api_instance = client.CoreV1Api()

body = {
    "metadata": {
        "labels": {
            "foo": "bar"}
        }
}

api_response = api_instance.patch_node("minikube", body)

pprint(api_response)
```

This adds a `foo` label with value `bar` to the node `minikube`.

To remove an existing label the body has to be changed to be:
```python
body = {
    "metadata": {
        "labels": {
            "foo": None}
        }
}
```

I'm really puzzed about why this is working. I read all the docs about the API request formats and I was pretty sure the original code was following the guidelines.
